consensus/state.go

```diff
@@ -1085,7 +1085,13 @@ func (cs *State) defaultDecideProposal(height int64, round int32) {
 
 	// Make proposal
 	propBlockID := types.BlockID{Hash: block.Hash(), PartSetHeader: blockParts.Header()}
-	proposal := types.NewProposal(height, cs.state.NextCoreChainLock.CoreBlockHeight, round, cs.ValidRound, propBlockID)
+
+	coreChainLockedBlockHeight := cs.state.InitialCoreChainLockedHeight
+	if cs.state.NextCoreChainLock.CoreBlockHeight > coreChainLockedBlockHeight {
+		coreChainLockedBlockHeight = cs.state.NextCoreChainLock.CoreBlockHeight
+	}
+
+	proposal := types.NewProposal(height, coreChainLockedBlockHeight, round, cs.ValidRound, propBlockID)
```

Comment:
This change is not good. The nextCoreChainLock is set the following way 
```
        if nextCoreChainLock == nil {
		nextCoreChainLock = &state.NextCoreChainLock
	}

	if nextCoreChainLock.CoreBlockHeight < lastCoreChainLock.CoreBlockHeight {
		nextCoreChainLock = lastCoreChainLock
	}
```

So you can see the nextCoreChainLock is always superior in height to the previous block and so would work fine. In this code if the nextCoreChainLock is inferior to the InitialCoreChainLockedHeight (I'm not sure how this could happen though) then the proposal is for the InitialCoreChainLockedHeight and not the last one as intended.

Reply:
Fixed in `NextCoreChainLockedHeight`